core/types/rollup_l1_cost.go

```diff
@@ -23,27 +23,32 @@ import (
 	"github.com/ethereum/go-ethereum/params"
 )
 
-type RollupGasData struct {
-	Zeroes, Ones uint64
+type L1CostData struct {
+	zeroes, ones uint64
 }
 
-func (r RollupGasData) DataGas(time uint64, cfg *params.ChainConfig) (gas uint64) {
-	gas = r.Zeroes * params.TxDataZeroGas
-	if cfg.IsRegolith(time) {
-		gas += r.Ones * params.TxDataNonZeroGasEIP2028
-	} else {
-		gas += (r.Ones + 68) * params.TxDataNonZeroGasEIP2028
+func NewL1CostData(data []byte) (out L1CostData) {
+	for _, byte := range data {
```

Comment:
nit: avoid using a variable name that is also a type

Reply:
fixed